### PR TITLE
✨(api) require at least one target when creating a tariff

### PR DESCRIPTION
**Pull request by boehm-e**
Reject tariff creation payloads with missing or empty targets, and update tariff tests/factories to use valid associated charge points.

### CODE REVIEW

File: src/api/tests/api/v1/routers/test_tariff.py [hunk 1]
```diff
@@ -179,40 +179,37 @@ def test_tariff_api_workflow(db_session, client_auth):
...
+    targets,
+):
+    """Test tariff creation rejects payloads without associated charge points."""
     start = datetime.now(timezone.utc) - timedelta(days=1)
     end = datetime.now(timezone.utc) + timedelta(days=1)
+    payload = _tariff_payload("tariff-1", start, end, ["FRS63E0001"])
+    if targets is None:
+        del payload["targets"]
+    else:
+        payload["targets"] = targets
```
Comment (jmaupetit) [inline src/api/tests/api/v1/routers/test_tariff.py]: I think you also need to test the `payload["targets"] = None` case (not only empty list or missing field in the payload).

File: src/api/tests/api/v1/routers/test_tariff.py [hunk 2]
```diff
@@ -179,40 +179,37 @@ def test_tariff_api_workflow(db_session, client_auth):
...
-        ).all()
-        == []
+        json=payload,
     )
 
-    response = client_auth.get(f"/tariff/{tariff_id}")
-    assert response.status_code == status.HTTP_200_OK
-    assert response.json()["id_pdc_itinerance"] == []
+    assert response.status_code == status.HTTP_422_UNPROCESSABLE_ENTITY
+    assert db_session.exec(select(Tariff)).all() == []
```
Comment (jmaupetit) [inline src/api/tests/api/v1/routers/test_tariff.py:225]: I would also test the response error details to check that we have raised the expected validation error.

File: src/api/tests/api/v1/routers/test_tariff.py
Comment (jmaupetit) [inline src/api/tests/api/v1/routers/test_tariff.py:262]: ditto or else, we have no idea why we get a 422.

File: src/api/qualicharge/factories/tariff.py [hunk 3]
```diff
@@ -70,7 +70,7 @@ class TariffCreateFactory(ModelFactory[TariffCreate]):
     """Tariff creation payload factory."""
 
     tariff = Use(TariffObjectFactory.build)
-    targets: list[str] = []
+    targets = Use(lambda: ["FRS63E0001"])
```
Comment (jmaupetit) [inline src/api/qualicharge/factories/tariff.py]: Why not using the same generator as in the `StatiqueFactory.id_pdc_itinerance`?

File: src/api/tests/models/test_tariff.py [hunk 4]
```diff
@@ -36,7 +36,7 @@ def test_tariff_create_factory():
     payload = TariffCreateFactory.build()
 
     assert payload.tariff.tariff_id
-    assert payload.targets == []
+    assert payload.targets == ["FRS63E0001"]
```
Comment (jmaupetit) [inline src/api/tests/models/test_tariff.py]: Not a big fan of hard-coded values as this. Following my previous comment, I would prefer testing that this field as at least one value and that this value matches the `id_pdc_itinerance` pattern.

**Review by jmaupetit — COMMENTED** [review]
That will solve many issues, thanks!

File: src/api/tests/api/v1/routers/test_tariff.py (hunk 1 above)
Comment (boehm-e) [reply src/api/tests/api/v1/routers/test_tariff.py]: 439d12a Ajoute le cas explicite targets: null, en plus des cas absent et liste vide.

File: src/api/tests/api/v1/routers/test_tariff.py (hunk 2 above)
Comment (boehm-e) [reply src/api/tests/api/v1/routers/test_tariff.py:225]: 05ea72d vérifie la localisation, le type et le message de chaque erreur de validation sur targets.

File: src/api/tests/api/v1/routers/test_tariff.py
Comment (boehm-e) [reply src/api/tests/api/v1/routers/test_tariff.py:262]: 8e074b1 vérifie que le 422 sur currency="USD" correspond bien à l’erreur Literal["EUR"].

File: src/api/qualicharge/factories/tariff.py (hunk 3 above)
Comment (boehm-e) [reply src/api/qualicharge/factories/tariff.py]: cac56ad remplace les identifiants codés en dur par un générateur repris de `StatiqueFactory.id_pdc_itinerance`

File: src/api/tests/models/test_tariff.py (hunk 4 above)
Comment (boehm-e) [reply src/api/tests/models/test_tariff.py]: d50cae2 remplace les assertions exactes par des vérifications de liste non vide et du format FRxxxE######.

**Review by jmaupetit — APPROVED** [review]
LGTM

**Review by jmaupetit — CHANGES_REQUESTED** [review]
Your commit title should be:

✨(api) require at least one target when creating a tariff

**Comment by jmaupetit** [comment]
@boehm-e thanks for updating the PR title, but you also need to update your commit title and force-push your branch in this PR.

**Review by jmaupetit — APPROVED** [review]
